File: app/src/main/java/com/example/wp/awesomemmz/PagerAdapter.java
package com.example.wp.awesomemmz;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.example.wp.awesomemmz.index.IndexFragment;
import com.example.wp.awesomemmz.skill.SkillFragment;
import com.example.wp.awesomemmz.star.StarFragment;

/**
 * Created by wp on 2019/1/30.
 */
public class PagerAdapter extends FragmentStatePagerAdapter {
	
	private String[] titles = {"Index", "Skill", "Star", "2021", "2022", "2023", "2024", "2025"};
	
	public PagerAdapter(FragmentManager fm) {
		super(fm);
	}
	
	@Override
	public Fragment getItem(int position) {
		if (position == 1) {
			return SkillFragment.newInstance("1", "2");
		} else if (position == 2) {
			return StarFragment.newInstance();
		}
		return new IndexFragment();
	}
	
	@Override
	public int getCount() {
		return titles.length;
	}
	
	@Nullable
	@Override
	public CharSequence getPageTitle(int position) {
		return titles[position];
	}
}

File: app/src/main/java/com/example/wp/awesomemmz/PagerAdapter.kt
package com.example.wp.awesomemmz

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import com.example.wp.awesomemmz.common.EmptyFragment
import com.example.wp.awesomemmz.index.IndexFragment
import com.example.wp.awesomemmz.javaz.JavazFragment
import com.example.wp.awesomemmz.other.OtherFragment
import com.example.wp.awesomemmz.skill.SkillFragment
import com.example.wp.awesomemmz.star.StarFragment

/**
 * Created by wp on 2019/1/30.
 */
class PagerAdapter(fm: FragmentManager?) : FragmentStatePagerAdapter(fm) {
    private val titles = arrayOf("Index", "Skill", "Star", "Others", "Java", "2023")

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> IndexFragment()
            1 -> SkillFragment.newInstance("1", "2")
            2 -> StarFragment.newInstance()
            3 -> OtherFragment()
            4 -> JavazFragment.newInstance(1)
            else -> EmptyFragment()
        }
    }

    override fun getCount(): Int {
        return titles.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles[position]
    }
}